core/web/resolver/query.go

```diff
@@ -230,3 +231,28 @@ func (r *Resolver) Node(ctx context.Context, args struct{ ID graphql.ID }) (*Nod
 
 	return NewNodePayloadResolver(&node, nil), nil
 }
+
+// JobProposal retrieves a job proposal by ID
+func (r *Resolver) JobProposal(ctx context.Context, args struct {
+	ID graphql.ID
+}) (*JobProposalPayloadResolver, error) {
+	if err := authenticateUser(ctx); err != nil {
+		return nil, err
+	}
+
+	id, err := strconv.ParseInt(string(args.ID), 10, 64)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Minor: would this be considered an `InputError`? @jkongie

Reply:
I personally didn't add an input error here due to how the REST controller is handling it, as an internal server error. Would love to know if we should treat it as an input error.

Reply:
We are good to leave the string conversion as a GQLError for now, but we should potentially standardise our error handling before continuing to work on more queries that we will have to refactor later.

For this specific error, in a future PR, I would lean towards a `NotFoundError`. An `ID` in GQL is just a wrapped string, but in this specific case it uses an int64. If the fetch call accepted a string instead, we would pass the argument in and it wouldn't match any id and return a `NotFoundError`